hazelcast/src/test/java/com/hazelcast/ringbuffer/impl/RingbufferAddAllReadManyStressTest.java

```diff
@@ -119,17 +126,28 @@ public void test(RingbufferConfig ringbufferConfig) {
         ProduceThread producer = new ProduceThread();
         producer.start();
 
+        long startTimeMillis = currentTimeMillis();
+
         sleepAndStop(stop, 60);
         logger.info("Waiting for completion");
 
         producer.assertSucceedsEventually();
-        consumer1.assertSucceedsEventually();
-        consumer2.assertSucceedsEventually();
-
-        logger.info(producer.getName() + " produced:" + producer.produced);
-
-        assertEquals(producer.produced, consumer1.seq);
-        assertEquals(producer.produced, consumer2.seq);
+        if (JitterRule.hasHiccupsOver(startTimeMillis, currentTimeMillis(), SECONDS.toNanos(ttlSeconds))) {
+            consumer1.assertTerminates();
```

Comment:
I would prefer an isolated solution, now measuring jitter is a direct part of this test's code. But at the same time, this may be seen a result of missing feature like ringbuffer-expiry-listener so this change makes sense from this perspective.

One potential issue may be this: What if there is another exception which has no jitter related root cause? How can i see that?

Reply:
> I would prefer an isolated solution, now measuring jitter is a direct part of this test's code.

Good idea, probably a maximum jitter tolerance junit rule would be a nice improvement (something similar to https://github.com/hazelcast/hazelcast/pull/13847 probably).

> But at the same time, this may be seen a result of missing feature like ringbuffer-expiry-listener so this change makes sense from this perspective.

Yes, having an expiry listener would allow the test to verify that everything produced was either consumed or expired.

> One potential issue may be this: What if there is another exception which has no jitter related root cause? How can i see that?

When jitter-larger-than-ttl is measured, then exceptions are hidden. However in the specific tests, when there is a single pause that is larger than configured TTL, then all bets are off so logging the exceptions (if some exist in `ConsumerThread#error`) most probably does not add any valuable output.

Reply:
It seems that more people start converging towards checking for jitter in the tests, and act accordingly. If we accept this as a solution (which quite honestly I have no objections with, given the current circumstances), I don't see why not broadening the scope and allowing the junit rules (https://github.com/hazelcast/hazelcast/pull/13847 AND/OR https://github.com/hazelcast/hazelcast/pull/16438) in the test codebase.

On the other hand some time-sensitive test failures, WITH jitter proved to be unrelated to jitter, and most likely would have been missed with these rules. So again, pros & cons.

CC'ing @mmedenjak 

At this point we have limited options.
- Allocate time to the build environment and tackle the root cause
- Work around the root cause by relaxing the assertions when jitter is detected. Maybe even introduce one more report, of how this relaxation affects the tests results over time.
- Close the issues & wait for a miracle :P

Reply:
I definitely appreciate all of the ideas around flaky test failures and slow environments but I feel like they are all addressing the symptoms instead of the cause. We've taken a look at the build environment before and I believe we've updated the JDK version (is that right @Holmistr @blazember ?). If that has changed, I'd like for us to take a look at the load again. To me it feels like otherwise we'll be start to add this test repeating or assertion-changing logic to many tests (e.g. most time-sensitive/TTL/max-idle tests).

Still, I might be overruled here, curious to hear what the rest of the team thinks, maybe we can discuss it on standup.

Reply:
We have updated the JDK. However, it doesn't really make any effect, since we're not passing any parameters to the Docker container. Jenkins plugin cannot do that currently. I'm going to work on this once we release 4.0 and things come down a bit :)